Hack to enable neovim support

malramsay64 opened the pull request:
This is a very first attempt at enabling neovim support for
vim-taskwarrior #132 . In my use the only functions that require interaction
are undo and delete, for these two functions there is a check
`has('nvim')` which changes execution from `!` to `terminal` and there is
also no `taskwarrior#refresh()` call, which is done using a BufEnter
autocmd.

There is likely a better solution that I don't yet know about.

malramsay64 committed:
Hack to enable neovim support

This is a very first attempt at enabling neovim support for
vim-taskwarrir. In my use the only functions that require interaction
are undo and delete, for these two functions there is a check
`has('nvim')` which changes execution from `!` to `terminal`. There is
also no `taskwarrior#refresh()` call, which is done in a BufEnter
autocmd.

malramsay64 committed:
Neovim support for sync

The refresh of the terminal occurred before sync had updated so any new
tasks were not showing up when syncing in neovim.

kalbasit commented on autoload/taskwarrior/action.vim:
```diff
@@ -85,23 +85,34 @@ function! taskwarrior#action#modify(mode)
+            endif
+        endif
+        if !has('nvim')
+            call taskwarrior#refresh()
```
I think you have in indentation problem here

kalbasit commented on autoload/taskwarrior/action.vim:
```diff
@@ -85,23 +85,34 @@ function! taskwarrior#action#modify(mode)
+        call taskwarrior#list()
+    else
+        execute 'terminal task '.taskwarrior#data#get_uuid().' delete'
+    endif
```
it's generally better if you start with the positive route `if has('nvim')`

malramsay64 replied:
Thanks for the feedback, I think I have fixed/improved the bits of code that you picked up.

malramsay64 committed:
Change to positive has('nvim') conditionals

All the conditionals in the check for nvim have been modified to check
for the positive rather than negative.

malramsay64 committed:
Additional windows/tabs no longer close

I am creating a new window every time there is a call to terminal, as
without it when there are multiple windows or tabs open the terminal
call will close the current window.
